src/syscheckd/db/include/fimDBHelper.hpp

```diff
@@ -146,7 +147,7 @@ namespace FIMDBHelper
     * @return 0 on success, another value otherwise.
     */
     template<typename T>
-    int updateItem(const std::string & tableName, const nlohmann::json & item)
```

Comment:
this function could be a void, and in case of error updateItem could throw an exception that is caught in the caller of this method. This is to avoid conditionals on success cases.